### PR TITLE
Added Tanger Outlet spider

### PR DESCRIPTION
**Pull request by rubyruins**
Done with the second one too :) @iandees @thekuiperbelt, could you please review?

### CODE REVIEW

File: locations/spiders/tanger_outlet.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+import scrapy
+from locations.items import GeojsonPointItem
+
+class TangerOutletSpider(scrapy.Spider):
+	name = "tanger_outlet"
```
Comment (iandees) [inline locations/spiders/tanger_outlet.py:6]: Can you switch to four spaces to match the rest of the repo please?

File: locations/spiders/tanger_outlet.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		storeurls = response.xpath('.//a[@title="Visit"]/@href').extract()
+		storeurls = [response.urljoin(i) for i in storeurls if 'location' in i]
+		for storeurl in storeurls:
+			yield scrapy.Request(storeurl, callback=self.parse_store)
+
+	def parse_store(self, response):
+		tempurl = '/' + response.url.split('/')[-2] + '/'
+		for i in response.xpath('.//ul[@id="usCenters"]/li[@class="allCenters"]/a/@href').extract():
+			if i == tempurl:
+				latitude = float(response.xpath('.//ul[@id="usCenters"]/li[@class="allCenters"]/@data-longitude').extract_first())
```
Comment (iandees) [inline locations/spiders/tanger_outlet.py:23]: It looks like the lat and lon aren't getting parsed properly. Take a look at the output from the test run:

https://placescraper-results.s3.amazonaws.com/map.html?show=https://placescraper-results.s3.amazonaws.com/ci/284655293/tanger_outlet/output.geojson

File: locations/spiders/tanger_outlet.py (hunk 1 above)
Comment (rubyruins) [reply locations/spiders/tanger_outlet.py:6]: Yep, will do that!

File: locations/spiders/tanger_outlet.py (hunk 2 above)
Comment (rubyruins) [reply locations/spiders/tanger_outlet.py:23]: Oops, my bad. I switched the latitude and longitude while storing their values. Will correct that!

**Comment by rubyruins** [comment]
@iandees I've changed the indentation and corrected the latitude / longitude error, is it parsing properly now? Thanks for pointing it out!

File: locations/spiders/tanger_outlet.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        storeurls = [response.urljoin(i) for i in storeurls if 'location' in i]
+        for storeurl in storeurls:
+            yield scrapy.Request(storeurl, callback=self.parse_store)
+
+    def parse_store(self, response):
+        tempurl = '/' + response.url.split('/')[-2] + '/'
+        for i in response.xpath('.//ul[@id="usCenters"]/li[@class="allCenters"]/a/@href').extract():
+            if i == tempurl:
+                latitude = float(response.xpath('.//ul[@id="usCenters"]/li[@class="allCenters"]/@data-latitude').extract_first())
+                longitude = float(response.xpath('.//ul[@id="usCenters"]/li[@class="allCenters"]/@data-longitude').extract_first())
```
Comment (iandees) [inline locations/spiders/tanger_outlet.py:24]: Looks like all the locations are sitting at the lat/lon of one store:

https://placescraper-results.s3.amazonaws.com/map.html?show=https://placescraper-results.s3.amazonaws.com/ci/284742598/tanger_outlet/output.geojson

Looking at the source for https://www.tangeroutlet.com/locations, it seems like you can get all the location information on that single page without having to make subsequent requests. Try putting the for loop that's in your current `parse_store()` function in the `parse()` method and see what happens.

Comment (rubyruins) [reply locations/spiders/tanger_outlet.py:24]: Ah, I completely didn't notice that all locations were exactly the same. I too thought of scraping everything from the main page too, with a couple different ways to select the divs . 

For some reason, whenever I targeted the `center-info-block` class it kept returning an empty list. I'll give it another shot and use a different combination of selectors :)

**Comment by rubyruins** [comment]
I changed my selectors to request everything from a single page. Is it working properly now? @iandees

**Comment by iandees** [comment]
Yep, looks good!

**Comment by rubyruins** [comment]
@iandees great! Well, this has been fun. I'd love to contribute in the future after I'm done with my other PRs.

**Pull request merged**